libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2959,3 +2966,20 @@ void IRGeneratorForStatements::setLocation(ASTNode const& _node)
 {
 	m_currentLocation = _node.location();
 }
+
+string IRGeneratorForStatements::linkerSymbol(TypeType const& _libraryTypeType) const
+{
+	ContractType const* contractType = dynamic_cast<ContractType const*>(_libraryTypeType.actualType());
+	solAssert(contractType, "");
+	return linkerSymbol(*contractType);
+}
+
+string IRGeneratorForStatements::linkerSymbol(ContractType const& _libraryType) const
+{
+	solAssert(_libraryType.contractDefinition().isLibrary(), "");
+
+	string escapedFullyQualifiedName = _libraryType.contractDefinition().fullyQualifiedName();
+	boost::replace_all(escapedFullyQualifiedName, "\"", "\\\"");
```

Comment:
What about backslashes?

Reply:
Good catch. I thought I could get away with limiting this to quotes but apparently not. I need full-blown escaping. `\n` won't work either and non-printable characters need escaping.

I found `util::escapeAndQuoteString()`. Is the output guaranteed to be safe for Yul/Solidity strings? I added a semantic test for this and also tested manually and it does seem to work.